### PR TITLE
feat: update data association in object merger

### PR DESCRIPTION
**Pull request by yukkysaito**
## Description
Update object merger
- delete the overlapped unknown object with the classified object.
- prioritize classified objects over unknown objects. 
- update data association. It is referenced by multi-object tracker

before : There are two overlapped buses. 
Peek 2022-06-23 19-30

after : 
Peek 2022-06-24 01-15

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
@miursh ping :pray:

**Comment by miursh** [comment]
@yukkysaito There are some copied source code added. I don't think It is a really good idea to create a copy of the codes. Is there any reason why you should do this?

**Comment by yukkysaito** [comment]
@miursh 
>There are some copied source code added. I don't think It is a really good idea to create a copy of the codes. Is there any reason why you should do this?

You are right.
I am planning to make a common library in another PR.

**Review by miursh — APPROVED** [review]
Almost LGTM
Just one question

File: perception/object_merger/src/object_association_merger/node.cpp [hunk 1]
```diff
@@ -12,77 +12,195 @@
...
+  }
+  return true;
+}
+
+bool isUnknownObjectOverlapped(
+  const autoware_auto_perception_msgs::msg::DetectedObject & unknown_object,
+  const autoware_auto_perception_msgs::msg::DetectedObject & known_object)
+{
+  constexpr double precision_threshold = 0.4;
+  constexpr double recall_threshold = 0.5;
```
Comment (miursh) [inline perception/object_merger/src/object_association_merger/node.cpp:85]: [Q] I'm not sure about these values, are constants ok? not parameter?

Comment (yukkysaito) [reply perception/object_merger/src/object_association_merger/node.cpp:85]: I will change it to a parameter. :+1:

Comment (yukkysaito) [reply perception/object_merger/src/object_association_merger/node.cpp]: @miursh fixed
[b381f1c](https://github.com/autowarefoundation/autoware.universe/pull/1001/commits/b381f1cef48efdaf238f2bd13a7a470e7924c7fe)

**Review by miursh — APPROVED** [review]
LGTM

**Pull request merged**